clvm_tools/cmds.py

```diff
@@ -227,10 +209,11 @@ def launch_tool(args, tool_name, default_stage=0):
     elif args.verbose or args.table:
         pre_eval_f = make_trace_pre_eval(log_entries)
 
-    run_script = getattr(args.stage, tool_name)
+    if hasattr(args.stage, tool_name):
+        arg_serialized = b"\xff" + program_serialized + arg_serialized
```

Comment:
I've always hate this construction because it depends upon the serialization format. Would it be too much to ask to use `Program.to(program, arg).as_bin()` (and factor the variables from above)?

So around 192 do

```
    program = binutils.assemble_from_ir(src_sex)
    program_serialized = program.as_bin()```

and similar for `arg`

Reply:
in the case where the program and args are passed in as hex programs, first converting it to `SExp` just to combine them and turn them back to serialized form can be quite expensive. some of the benchmarks run large programs or blocks.

Right now I simplified all these combinations by always keeping the program and arg in serialized form, because that's how they're passed to the rust API.

If you pass in the programs in text-form, they're first parsed then serialized.

If you run against the python back-end, they are de-serialized into `SExp` (but the python back-end would be slow anyway, for large programs, so it's unlikely to be a problem there).

When running the `run` tool (or stage 2) where a launcher/wrapper program is used, we need to combine the serialized forms.

in an ideal world, we could keep the program and args in the format they were passed in, to avoid all redundant conversions across the combinations of hex or text input. rust or python backend, and launcher program or not.

However, that would add a lot of complexity with very little gain (afaict). I think the case that really matters isn't slow is hex-input and rust back-end, so that's what I optimized for.

Reply:
I don't see a simple and practical way to use the proper serializer to build this cons-box. I might be missing something.

Reply:
Got it. We need a `Program`/`SerializedProgram` hybrid to do this right. Someday.